bundle/jsky.app.ot/src/main/java/jsky/app/ot/gemini/editor/targetComponent/TelescopePosTableWidget.java

```diff
@@ -917,15 +917,20 @@ private void _resetTable(final TargetEnvironment newEnv) {
     void updatePrimaryStar() {
         if (_env == null || !OTOptions.isEditable(_obsComp.getProgram(), _obsComp.getContextObservation())) return;
 
-        // TODO: This method must be tested in greater detail, since custom guiding is required for offset positions.
+        final TargetEnvironment env = _dataObject.getTargetEnvironment();
         final Option<SPTarget> targetOpt = getSelectedPos();
-        if (targetOpt.isDefined()) {
+        final Option<IndexedGuideGroup> iggOpt = targetOpt.flatMap(this::getTargetGroup).orElse(getSelectedGroup());
+        final boolean primaryGroupIsSelected = iggOpt.exists(igg -> igg.group().equals(env.getOrCreatePrimaryGuideGroup()));
```

Comment:
I was wondering about that myself.
I'll just make the change now and then merge this sucker.